www/apps/resources/app/recipes/marketplace/examples/vendors/page.mdx

```diff
@@ -0,0 +1,1344 @@
+    this.addSql('CREATE UNIQUE INDEX IF NOT EXISTS "IDX_vendor_admin_email_unique" ON "vendor_admin" (email) WHERE deleted_at IS NULL;');
+    this.addSql('CREATE INDEX IF NOT EXISTS "IDX_vendor_admin_vendor_id" ON "vendor_admin" (vendor_id) WHERE deleted_at IS NULL;');
+
+    this.addSql('alter table if exists "vendor_admin" add constraint "vendor_admin_vendor_id_foreign" foreign key ("vendor_id") references "vendor" ("id") on update cascade;');
+  }
+
+  async down(): Promise<void> {
+    this.addSql('alter table if exists "vendor_admin" drop constraint if exists "vendor_admin_vendor_id_foreign";');
+
+    this.addSql('drop table if exists "vendor" cascade;');
+
+    this.addSql('drop table if exists "vendor_admin" cascade;');
+  }
+
+}
+```
+
+Next, create the main service of the module at `src/modules/marketplace/service.ts` with the following content:
+
+export const serviceHighlights = [
+  ["11", "createVendors", "Override the `createVendors` method generated by `MedusaService`."]
+]
+
+```ts title="src/modules/marketplace/service.ts" highlights={serviceHighlights}
+import { MedusaService } from "@medusajs/utils"
+import Vendor from "./models/vendor"
+import VendorAdmin from "./models/vendor-admin"
+import { CreateVendorData, VendorData } from "./types"
+
+class MarketplaceModuleService extends MedusaService({
+  Vendor,
+  VendorAdmin
+}) {
+  // @ts-expect-error
+  async createVendors(
+    data: CreateVendorData[]
+  ): Promise<VendorData[] | VendorData> {
+    const input = (Array.isArray(data) ? data : [data]).map(
+      (vendor) => {
+        if (!vendor.handle) {
+          vendor.handle = vendor.name.toLowerCase().replace(/\s/g, "-")
+        }
+
+        return vendor
+      }
+    )
+
+    return await super.createVendors(input)
+  }
+}
+
+export default MarketplaceModuleService
+```
+
+The service extends the [service factory](!docs!/advanced-development/modules/service-factory), which provides basic data-management features. It overrides the `createVendors` method to generate a vendor's `handle` if it doesn’t have one on creation.
+
+The service uses types in its method. Create these types in the `src/modules/marketplace/types/index.ts` file:
+
+```ts title="src/modules/marketplace/types/index.ts"
+export type VendorData = {
+  id: string
+  name: string
+  logo?: string
+  handle?: string
+}
+
+export type CreateVendorData = Omit<VendorData, "id">
+
+```
+
+After that, create the module definition at `src/modules/marketplace/index.ts` with the following content:
+
+```ts title="src/modules/marketplace/index.ts"
+import { Module } from "@medusajs/utils"
+import MarketplaceModuleService from "./service"
+
+export const MARKETPLACE_MODULE = "marketplaceModuleService"
```

Comment:
We aligned before on making it a convention for custom module to be `{moduleName}ModuleService`.

Reply:
That would be the case for the module service, but this should not be the module key. But maybe this is not something to address in documentation. It will likely need some work/renamings in the framework first.

Reply:
Cool, from what Adrien mentioned they all should be the same name

Reply:
Will create a ticket, so we can address later 👍